alerting/src/main/kotlin/org/opensearch/alerting/resthandler/RestIndexMonitorAction.kt

```diff
@@ -106,23 +124,68 @@ class RestIndexMonitorAction : BaseRestHandler() {
                     Monitor.MonitorType.QUERY_LEVEL_MONITOR -> {
                         triggers.forEach {
                             if (it !is QueryLevelTrigger) {
-                                throw (IllegalArgumentException("Illegal trigger type, ${it.javaClass.name}, for query level monitor"))
+                                throw (IllegalArgumentException("Illegal trigger type, ${it.javaClass.name}, for query level monitor."))
+                            }
+                            if (!isValidName(it.name)) {
+                                throw IllegalArgumentException(
+                                    "Invalid trigger name [${it.name}]. " +
+                                        "Trigger Name should be alphanumeric (4-50 chars) starting with letter or underscore."
+                                )
+                            }
+                            it.actions.forEach { action ->
+                                val destinationId = action.destinationId
+                                if (!isValidId(destinationId)) {
+                                    throw IllegalArgumentException(
+                                        "Invalid destination ID [$destinationId]. " +
+                                            "Destination ID should be alphanumeric string with +, /, _, or - characters only."
+                                    )
+                                }
                             }
                         }
                     }
 
                     Monitor.MonitorType.BUCKET_LEVEL_MONITOR -> {
                         triggers.forEach {
                             if (it !is BucketLevelTrigger) {
-                                throw IllegalArgumentException("Illegal trigger type, ${it.javaClass.name}, for bucket level monitor")
+                                throw IllegalArgumentException("Illegal trigger type, ${it.javaClass.name}, for bucket level monitor.")
+                            }
+                            if (!isValidName(it.name)) {
```

Comment:
Discussed with @vikhy-aws offline that it would be helpful to add a validate function to the [trigger interface](https://github.com/opensearch-project/common-utils/blob/main/src/main/kotlin/org/opensearch/commons/alerting/model/Trigger.kt#L11) to validate the various fields rather than adding each check here. Specific trigger types (e.g., [bucket level trigger](https://github.com/opensearch-project/common-utils/blob/main/src/main/kotlin/org/opensearch/commons/alerting/model/BucketLevelTrigger.kt)) can then override the interface's validate function as needed to accommodate their specific validation scenarios.